src/opera/commands/deploy.py

```diff
@@ -37,24 +47,50 @@ def add_parser(subparsers):
 
 
 def _parser_callback(args):
+    if args.resume and args.force:
+        print("The -r/--resume and -f/--force "
+              "options cannot be used together.")
+        return 0
+
     if args.instance_path and not path.isdir(args.instance_path):
-        raise argparse.ArgumentTypeError("Directory {0} is not a valid path!".format(args.instance_path))
+        raise argparse.ArgumentTypeError("Directory {0} is not a valid path!"
+                                         .format(args.instance_path))
 
     storage = Storage.create(args.instance_path)
 
     if args.workers < 1:
         print("{0} is not a positive number!".format(args.workers))
         return 1
 
-    # TODO(@tadeboro): This should be part of the init command that we do not
-    # have yet.
+    if storage.exists("instances"):
+        if args.resume:
+            print("The resume deploy option might have unexpected "
+                  "consequences on the already deployed blueprint.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+        elif args.force:
+            print("The force deploy option might have unexpected "
+                  "consequences on the already deployed blueprint.")
+            question = prompt_yes_no_question()
+            if question:
+                storage.remove("instances")
+            else:
+                return 0
+        else:
+            print("The instance model already exists. Use --resume to "
+                  "continue or --force to ignore current deployment state "
+                  "and start over the deployment.")
+            return 0
+
```

Comment:
This whole block needs to be restructured to reside in `deploy()`, because it's application logic, not argument validation and parsing. A good guideline is that if the CLI is the only one that could possibly do this, it goes into the callback, otherwise it needs to be in the library method.

Here, removing the instances file needs to be in the library function, but the prompts mustn't be there. You need to think about how you want to design the CLI and library - here is my idea:

* Make the CLI user choose between resuming and forcing.
  * Resuming ignores any errors, forcing deletes the instance file (==sets a flag, see next line)
* Make the library user choose whether to delete a previous deployment by adding a `delete_existing` boolean parameter. This makes the library user explicitly choose whether to silently ignore an existing deployment or delete it.

The only thing you need to do (to keep me happy :)) is to keep the library interface usable, and here that includes not having special behaviour for the CLI.

Reply:
Indeed, the functionality should be also available for the xOperaAPI and not only for CLI. Nice catch @sstanovnik.

Reply:
Removing instances has been moved to the library method and prompts and CLI arguments were kept where they had been before. I've also used the `delete_existing` boolean parameter.

Reply:
and what delete existing will do?

a) full undeploy and delete .opera ?
b) only delete .opera temp folder?

Reply:
`delete_existing` would delete `instances` subfolder that is contained in `.opera` folder which means that it would erase the created instance model.

Reply:
my personal problem is that `delete_existing` is not self-explanatory. Probably we already had an idea of  `delete_state` or  remove the current state with `clean` or `state clean`, etc.

Reply:
Aha I see. No worries - the parameter can simply be renamed to `delete_instances` or `delete_instance_model` or `delete_state` to make it clearer.  `opera deploy --force` deletes the created instance model and starts the deployment again. However, if we would also like to have a new argument (i.e. `--clean-state` like you mentioned), which would only delete instance model (i.e. state) we can do that in a this/separate PR but the usage of this clean option then needs to be explained further ad more in detail.